sycl/include/sycl/accessor.hpp

```diff
@@ -2136,7 +2136,8 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   template <access::target AccessTarget_ = AccessTarget,
             typename = detail::enable_if_t<
                 (AccessTarget_ == access::target::host_buffer) ||
-                (AccessTarget_ == access::target::host_task)>>
+                (AccessTarget_ == access::target::host_task) ||
+                (AccessTarget_ == access::target::device)>>
 #if SYCL_LANGUAGE_VERSION >= 202001
```

Comment:
Since we are now in the ABI/API breaking window and this is breaking API, we might as well remove this guard while we are at it.